src/app/reporting/Engine.cpp

```diff
@@ -535,14 +565,14 @@ void Engine::UpdateReadHandlerDirty(ReadHandler & aReadHandler)
     bool intersected = false;
     for (auto clusterInfo = aReadHandler.GetAttributeClusterInfolist(); clusterInfo != nullptr; clusterInfo = clusterInfo->mpNext)
     {
-        for (auto path = mpGlobalDirtySet; path != nullptr; path = path->mpNext)
-        {
+        mGlobalDirtySet.ForEachActiveObject([&](auto * path) {
             if (path->IsAttributePathSupersetOf(*clusterInfo) || clusterInfo->IsAttributePathSupersetOf(*path))
             {
                 intersected = true;
-                break;
+                return Loop::Break;
             }
-        }
+            return Loop::Continue;
+        });
     }
```

Comment:
Can't we break out of the outer loop if `intersected` has become true?

Reply:
yes, should break the outloop when intersected has become true. updated the code